lib/Net/SAML2/Binding/Redirect.pm

```diff
@@ -214,22 +233,27 @@ sub verify {
 
     $sig = decode_base64($sig);
 
-    if ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256') {
-        $rsa_pub->use_sha256_hash;
-    } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha224') {
-        $rsa_pub->use_sha224_hash;
-    } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha384') {
-        $rsa_pub->use_sha384_hash;
-    } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha512') {
-        $rsa_pub->use_sha512_hash;
-    } elsif ($sigalg eq 'http://www.w3.org/2000/09/xmldsig#rsa-sha1') {
-        $rsa_pub->use_sha1_hash;
-    } else {
-        die "Unsupported Signature Algorithim: $sigalg";
+    foreach my $crt (@{ $self->cert }) {
+        for my $use (keys %{$crt}) {
+            my $cert = Crypt::OpenSSL::X509->new_from_string($crt->{$use});
+            my $rsa_pub = Crypt::OpenSSL::RSA->new_public_key($cert->pubkey);
+
+            if ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256') {
+                $rsa_pub->use_sha256_hash;
+            } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha224') {
+                $rsa_pub->use_sha224_hash;
+            } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha384') {
+                $rsa_pub->use_sha384_hash;
+            } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha512') {
+                $rsa_pub->use_sha512_hash;
+            } elsif ($sigalg eq 'http://www.w3.org/2000/09/xmldsig#rsa-sha1') {
+                $rsa_pub->use_sha1_hash;
+            } else {
+                die "Unsupported Signature Algorithim: $sigalg";
```

Comment:
As we are looping over certs, I'm not sure if we should die when we don't support things. Perhaps we do support others. I think a warn is a better option. Or perhaps let the caller decide if they want to die here.